src/Config/Requirement/Violation/RequirementViolationList.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/*
+ * This file is part of the Manala package.
+ *
+ * (c) Manala <contact@manala.io>
+ *
+ * For the full copyright and license information, please refer to the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Manala\Config\Requirement\Violation;
+
+use Manala\Config\Requirement\Common\RequirementLevel;
+
+/**
+ * Iterable collection of requirement violations.
+ *
+ * @author Xavier Roldo <xavier.roldo@elao.com>
+ */
+class RequirementViolationList implements \Iterator, \ArrayAccess, \Countable
```

Comment:
Eh eh, i didn't really understand the \ArrayObject SPL class but i think i will use it intensively in the future (your recommendations about it are welcome !).